app/src/main/java/com/asfoundation/wallet/ui/gamification/GamificationInteractor.kt

```diff
@@ -53,49 +53,29 @@ class GamificationInteractor @Inject constructor(
     packageName: String, amount: BigDecimal,
     promoCodeString: String?, currency: String?
   ): Single<ForecastBonusAndLevel> {
-    return getCurrentPromoCodeUseCase()
-      .flatMap { promoCode ->
-        defaultWallet.find()
-          .flatMap { wallet: Wallet ->
-            gamification.getEarningBonus(
-              wallet.address, packageName, amount,
-              promoCodeString, currency
-            ).flatMap { appcBonusValue: ForecastBonus ->
-              Single.zip(
-                conversionService.getAppcToLocalFiat(appcBonusValue.amount.toString(), 18),
-                gamification.getUserBonusAndLevel(wallet.address, promoCode.code)
-              ) { localCurrency: FiatValue, userBonusAndLevel: ForecastBonusAndLevel ->
-                map(appcBonusValue, localCurrency, userBonusAndLevel)
-              }
-            }
+    return defaultWallet.find()
+      .flatMap { wallet: Wallet ->
+        gamification.getEarningBonus(
+          wallet.address, packageName, amount,
+          promoCodeString, currency
+        ).flatMap { appcBonusValue ->
+          conversionService.getAppcToLocalFiat(appcBonusValue.amount.toString(), 18).flatMap { fiatValue ->
```

Comment:
Instead of using a flatMap that returns a new Single, you can have a rxJava map. This way you can use the same Single by mapping only it's content